docs/proposals/20201125-jaeger-traces-store.md

```diff
@@ -0,0 +1,161 @@
+
+#### Single Jaeger instance - soft multitenancy
+
+Soft multitenancy in Jaeger is not supported at the moment. The progress is tracked in [jaegertracing/jaeger/3427](https://github.com/jaegertracing/jaeger/issues/3427).
+
+Here are few issues that needs to be addressed. The list might not be complete:
+* Service and operation name API. The service query API nor storage does not expose/store labels.
+* Dependency/service architecture diagram. To support this feature dependency schema, query, collector and Spark aggregation would have to change.
+* Data retention/TTL configurable pre tenant. The data retention in Elasticsearch is configurable per index.
+
+### Jaeger Query
+
+There are two ways users could access data stored in Jaeger:
+1. Expose only Jaeger Query API in the Observatorium API.
+2. Expose Jaeger UI outside the cluster.
+
+#### Expose only Jaeger Query API
+
+Jaeger query API would be exposed in the Observatorium API and users would have to deploy Jaeger UI and use Observatorium API as a data source.
+
+Jaeger query exposes multiple HTTP and gRPC APIs. The Jaeger UI uses HTTP API. This API does not have OpenAPI schema, and it is only defined in [http_handler.go](https://github.com/jaegertracing/jaeger/blob/0dd3e2da0579caed9e24ad2782a1f638ad63214d/cmd/query/app/http_handler.go#L119). The API uses this [data model](https://github.com/jaegertracing/jaeger/blob/9cd7a7ec1aa43b24a8a970eb5b393ca2ffd98a5d/model/json/model.go#L52).
+
+#### Expose Jaeger UI
+
+In addition to exposing Jaeger query API the Observatorium could deploy Jaeger UI as well. At the moment Observatorium does not expose any UI and it is expected that users deploy UI locally.
+
+## Alternatives
+
+### Grafana Tempo
+
+[Grafana Tempo](https://github.com/grafana/tempo) is an alternative storage that Observatorium project could use to store traces. The project is inspired and similar to the Prometheus TSDB, Thanos and Grafana Loki, therefore could it could be a good fit for Obeservatorium. The storage is split into two parts: 1. low retention requiring fast local disk (e.g. SSD) to store recent data and 2. object store (S3 compatible) to store historical data. Tempo is a distributed system with [multiple components](https://grafana.com/docs/tempo/latest/operations/architecture/): distributor, ingester, compactor, storage, querier. Distributor uses parts of the OpenTelemetry collector codebase to receive data in multiple formats (OTLP, Jaeger, Zipkin).
```

Comment:
Might be worth also mentioning it support multitenancy (https://grafana.com/docs/tempo/latest/configuration/multitenancy/) which is one of the features we're actually missing for Jaeger.

Reply:
added multitenant to the text